gdmix-workflow/src/gdmixworkflow/random_effect_workflow_generator.py

```diff
@@ -74,41 +74,23 @@ def get_partition_job(self, random_effect_config_obj, prev_model_name):
             "--outputPartitionListFile": partition_list_file,
-            "--predictionScoreColumnName": random_effect_config_obj.prediction_score_column_name,
+            "--predictionScoreColumnName": random_effect_config_obj.gdmix_config['prediction_score_column_name'],
             "--trainingScoreDir": prev_train_score_dir,
-            "--validationScoreDir": prev_validation_score_dir
-        }
-        return (GDMIX_SPARKJOB, "{}-partition".format(random_effect_name),
-                "com.linkedin.gdmix.data.DataPartitioner", params)
+            "--validationScoreDir": prev_validation_score_dir}
+        return GDMIX_SPARKJOB, f"{random_effect_name}-partition", "com.linkedin.gdmix.data.DataPartitioner", params
 
-    def get_train_job(self, random_effect_config_obj):
+    def get_train_job(self, name, random_effect_config_obj, gdmix_config):
         """ Get tfjob training job.
         Return: [job_type, job_name, "", job_params]
         """
-        random_effect_name = random_effect_config_obj.name
-        params = {STAGE: RANDOM_EFFECT}
-        # get params from config
-        flatten_config_obj(params, random_effect_config_obj)
-        # adjust param keys
-        params.pop("name")
-        params[PREDICTION_SCORE_COLUMN_NAME] = params.pop(PREDICTION_SCORE_COLUMN_NAME)
-        # adjust training/validation data and metadata path as the output of partition job
-        training_data_dir, validation_data_dir, metadata_file, partition_list_file = \
-        self.get_train_input_paths(random_effect_name)
-        params[TRAINING_DATA_DIR] = training_data_dir
-        params[VALIDATION_DATA_DIR] = validation_data_dir
-        params[METADATA_FILE] = metadata_file
-        params[PARTITION_LIST_FILE] = partition_list_file
-        # add output params
-        output_model_dir, training_score_dir, validation_score_dir = \
-        self.get_train_output_paths(random_effect_name)
-
-        params[OUTPUT_MODEL_DIR] = output_model_dir
-        params[TRAINING_SCORE_DIR] = training_score_dir
-        params[VALIDATION_SCORE_DIR] = validation_score_dir
-
-        params = prefix_dash_dash(params)
-        return (GDMIX_TFJOB, "{}-tf-train".format(random_effect_name), "", params)
+        training_data_dir, validation_data_dir, metadata_file, partition_list_file = self.get_train_input_paths(name)
+        output_model_dir, training_score_dir, validation_score_dir = self.get_train_output_paths(name)
+
+        gdmix_params = Params(**gdmix_config, stage=RANDOM_EFFECT, partition_list_file=partition_list_file,
+                              training_score_dir=training_score_dir, validation_score_dir=validation_score_dir)
+        re_params = replace(REParams(**random_effect_config_obj, output_model_dir=output_model_dir),
+                            training_data_dir=training_data_dir, validation_data_dir=validation_data_dir, metadata_file=metadata_file)
+        return GDMIX_TFJOB, f"{name}-tf-train", "", (gdmix_params, re_params)
```

Comment:
What's the gdmix_params and re_params? If the turn type changes, can you update the method documentation?

Reply:
There are typed parameters. The method documentation did not have any information for the param types.

Reply:
Previous the it returns `[job_type, job_name, "", job_params]` and the `job_params` is a simple dict, I think now the param, e.g., (gdmix_params, re_params), has more complex types? could you update the documentation? 

BTW, what the difference between gdmix_params and re_params?

Reply:
Those are the parameters in the trainers:

from gdmix.models.custom.random_effect_lr_lbfgs_model import REParams
from gdmix.params import Params